org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/preferences/DebuggerPreferencePage.scala

```diff
@@ -59,6 +77,16 @@ object DebuggerPreferencePage {
   val FILTER_SETTER = BASE_FILTER + Setter
   val FILTER_DEFAULT_GETTER = BASE_FILTER + DefaultGetter
   val FILTER_FORWARDER = BASE_FILTER + Forwarder
+
+  val HotCodeReplaceEnabled = "org.scala-ide.sdt.debug.hcr.enabled"
+  val NotifyAboutFailedHcr = "org.scala-ide.sdt.debug.hcr.notifyFailed"
+  val NotifyAboutUnsupportedHcr = "org.scala-ide.sdt.debug.hcr.notifyUnsupported"
+
+  // like in Java, it can lead to broken debug session but we have it for consistency
+  // with Java's hcr implementation
```

Comment:
This comment should be placed at the logic that replaces code with errors, not here in the preference page.